### PR TITLE
Introduce polling pallet

### PR DESCRIPTION
**Pull request by gavofyork**
CC @jacogr

### CODE REVIEW

File: runtime/polkadot/src/poll.rs [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+		TooLate,
+	}
+}
+
+decl_module! {
+	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+		type Error = Error<T>;
+
+		fn deposit_event() = default;
+
```
Comment (jacogr) [inline runtime/polkadot/src/poll.rs:73]:
```suggestion
		fn deposit_event() = default;

		/// The end of the poll.
		const End: T::BlockNumber = T::End::get();

```

^^^ Untested :)

File: runtime/polkadot/src/poll.rs [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+		/// Create a new crowdfunding campaign for a parachain slot deposit for the current auction.
```
Comment (shawntabrizi) [inline runtime/polkadot/src/poll.rs:74]: copy pasta

File: runtime/polkadot/src/poll.rs [hunk 3]
```diff
@@ -0,0 +1,235 @@
+// Copyright 2017-2020 Parity Technologies (UK) Ltd.
```
Comment (bkchr) [inline runtime/polkadot/src/poll.rs:1]:
```suggestion
// Copyright 2020 Parity Technologies (UK) Ltd.
```

File: runtime/polkadot/src/lib.rs [hunk 4]
```diff
@@ -941,6 +942,16 @@ impl frame_support::traits::OnRuntimeUpgrade for CustomOnRuntimeUpgrade {
 	}
 }
 
+parameter_types! {
+	pub const PollEnd: u32 = 100_000;
```
Comment (kaichaosun) [inline runtime/polkadot/src/lib.rs:946]:
```suggestion
	pub const PollEnd: BlockNumber = 100_000;
```

File: runtime/polkadot/src/poll.rs [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+	use super::*;
+
+	use frame_support::{
+		assert_ok, assert_noop, impl_outer_origin, parameter_types, weights::Weight,
+		ord_parameter_types,
+	};
+	use sp_core::H256;
+	use frame_system::{EnsureSignedBy, EnsureOneOf, EnsureRoot};
+	// The testing primitives are very useful for avoiding having to work with signatures
+	// or public keys. `u64` is used as the `AccountId` and no `Signature`s are required.
```
Comment (kianenigma) [inline runtime/polkadot/src/poll.rs:116]:
```suggestion
```

File: runtime/polkadot/src/poll.rs [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+		Perbill, testing::Header, traits::{BlakeTwo256, IdentityLookup},
+	};
+
+	impl_outer_origin! {
+		pub enum Origin for Test where system = frame_system {}
+	}
+
+	// For testing the pallet, we construct most of a mock runtime. This means
+	// first constructing a configuration type (`Test`) which `impl`s each of the
+	// configuration traits of pallets we want to use.
```
Comment (kianenigma) [inline runtime/polkadot/src/poll.rs:127]:
```suggestion
```

**Pull request merged**